enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/ChartTypeSelector.tsx

```diff
@@ -1,48 +1,20 @@
+import { useQuestionVisualization } from "embedding-sdk/components/private/InteractiveQuestion/hooks/use-question-visualization";
 import CS from "metabase/css/core/index.css";
-import {
-  ChartTypeSettings,
-  useChartTypeVisualizations,
-} from "metabase/query_builder/components/chart-type-selector";
+import { ChartTypeSettings } from "metabase/query_builder/components/chart-type-selector";
 import type { CardDisplayType } from "metabase-types/api";
 
-import { useInteractiveQuestionContext } from "../context";
-
-// Public facing hook so users can create their own selectors for their UI.
-// Will document once this PR is ready to go.
-export const useChartTypeSelectors = () => {
-  const { question, queryResults, updateQuestion } =
-    useInteractiveQuestionContext();
-
-  const {
-    selectedVisualization,
-    updateQuestionVisualization,
-    sensibleVisualizations,
-    nonSensibleVisualizations,
-  } = useChartTypeVisualizations({
-    question,
-    result: queryResults?.[0],
-    onUpdateQuestion: updateQuestion,
-  });
-
-  return {
-    selectedVisualization,
-    updateQuestionVisualization,
-    sensibleVisualizations,
-    nonSensibleVisualizations,
-  };
-};
+import { useSensibleVisualizations } from "../hooks/use-sensible-visualizations";
 
 export const ChartTypeSelector = ({
   onChange,
 }: {
   onChange?: (display: CardDisplayType) => void;
 }) => {
-  const {
-    selectedVisualization,
-    updateQuestionVisualization,
-    sensibleVisualizations,
-    nonSensibleVisualizations,
-  } = useChartTypeSelectors();
+  const { sensibleVisualizations, nonSensibleVisualizations } =
+    useSensibleVisualizations();
+
+  const { selectedVisualization, updateQuestionVisualization } =
+    useQuestionVisualization();
```

Comment:
I like this grouping of hooks! It looks like we're always using the two hooks in tandem in this PR though, do we intend to expose these hooks publicly at some point therefore we split it?

Reply:
yeah that's the idea!